compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIESplitLogicalObjFifos.cpp

```diff
@@ -59,9 +59,9 @@ LogicalResult collectSplittingDims(
     auto iter = std::find_if(memrefShape.begin(), memrefShape.end(),
                              [](int64_t size) { return size > 1; });
     size_t objFifoSplitDim = std::distance(memrefShape.begin(), iter);
-    // If all dimensions are unit (1), no splitting can be done, so continue to
-    // the next pair.
-    if (objFifoSplitDim >= memrefShape.size()) continue;
+    // Only consider splitting on dimensions 0 and 1.
+    if (objFifoSplitDim >= 2) continue;
```

Comment:
It would be better to just assume that we can only split on the outer dimension (dim 0) and skip/continue if that one is a unit dimension (no splitting). The packing permutation can/have been adjusted so that this should work and it shouldn't have any performance impact.